jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -98,11 +95,18 @@ protected WebSocketExtensionRegistry getExtensionRegistry()
 
     protected HttpClient getHttpClient()
     {
-        return coreClient.getHttpClient();
+        return getWebSocketCoreClient().getHttpClient();
     }
 
-    protected WebSocketCoreClient getWebSocketCoreClient() throws Exception
+    protected WebSocketCoreClient getWebSocketCoreClient()
     {
+        if (coreClient == null)
+        {
+            coreClient = coreClientFactory.get();
+            this.coreClient.getHttpClient().setName("Javax-WebSocketClient@" + Integer.toHexString(this.coreClient.getHttpClient().hashCode()));
```

Comment:
The Supplier controls the creation of the WebSocketCoreClient, don't force HttpClient.name here.

Reply:
ack